src/Claimer.sol

```diff
@@ -1,70 +1,102 @@
+        uint256 _maximumFee,
+        uint256 _timeToReachMaxFee,
+        UD2x18 _maxFeePortionOfPrize
+    ) {
         prizePool = _prizePool;
-        decayConstant = LinearVRGDALib.getDecayConstant(_priceDeltaScale);
-        targetPrice = _targetPrice;
+        maxFeePortionOfPrize = _maxFeePortionOfPrize;
+        decayConstant = LinearVRGDALib.getDecayConstant(LinearVRGDALib.getMaximumPriceDeltaScale(_minimumFee, _maximumFee, _timeToReachMaxFee));
+        minimumFee = _minimumFee;
     }
 
-    /***
-     * @return Fees earned
-     */
     function claimPrizes(
-        IVault _vault,
-        address[] calldata _winners,
-        uint8[] calldata  _tiers,
-        uint256 _minFees,
+        uint256 drawId,
+        Claim[] calldata _claims,
         address _feeRecipient
-    ) external returns (uint256) {
-        require(_winners.length == _tiers.length, "data mismatch");
-        require(_winners.length > 0, "no winners passed");
-        uint256 estimatedFees = _estimateFees(_winners.length);
-        require(estimatedFees >= _minFees, "insuff fee");
-
-        uint256 feePerClaim = estimatedFees / _winners.length;
-        uint256 actualFees;
-
-        for (uint i = 0; i < _winners.length; i++) {
-            if (_vault.claimPrize(_winners[i], _tiers[i], _winners[i], uint96(feePerClaim), _feeRecipient) != 0) {
-                actualFees += feePerClaim;
-            }
-        }
-
-        return actualFees;
-    }
-
-    function estimateFees(uint256 _claimCount) external returns (uint256) {
-        return _estimateFees(_claimCount);
-    }
-
-    function _estimateFees(uint256 _claimCount) internal returns (uint256) {
-
+    ) external returns (uint256 claimCount, uint256 totalFees) {
+        // console2.log("STARTING....");
+        
+        // The below values can change if the draw changes, so we'll cache them then add a protection below to ensure draw id is the same
         uint256 drawPeriodSeconds = prizePool.drawPeriodSeconds();
+        // console2.log("estimatedPrizeCount", prizePool.estimatedPrizeCount());
         SD59x18 perTimeUnit = LinearVRGDALib.getPerTimeUnit(prizePool.estimatedPrizeCount(), drawPeriodSeconds);
-        uint256 sold = prizePool.claimCount();
         uint256 elapsed = block.timestamp - (prizePool.lastCompletedDrawStartedAt() + drawPeriodSeconds);
+        // console2.log("elapsed", elapsed);
+
+        // compute the maximum fee based on the smallest prize size.
+        uint256 maxFee = _computeMaxFee();
 
-        uint256 estimatedFees;
+        // console2.log("maxFee", maxFee);
 
-        for (uint i = 0; i < _claimCount; i++) {
-            estimatedFees += LinearVRGDALib.getVRGDAPrice(targetPrice, elapsed, sold+i, perTimeUnit, decayConstant);
+        for (uint i = 0; i < _claims.length; i++) {
+            // ensure that the vault didn't complete the draw
+            if (prizePool.lastCompletedDrawId() != drawId) {
+                revert DrawInvalid();
```

Comment:
Since we can't claim if the `drawId` passed is not the last completed draw, shouldn't we just not pass the `drawId` at all and just claim for the last completed draw?

Reply:
No- because if the vault completes the draw then the `prizePool.claimCount()` will return to zero, but the algorithm will think a ton of time has elapsed.  This means the fees will be maxed out unnecessarily so

This was my struggle with the claimer: that ideally we should memoize the prizePool draw params, but we also need to protect ourselves in the event of a malicious Vault (they can't be trusted)